src/DonationForms/Listeners/StoreCustomFields.php

```diff
@@ -21,20 +25,56 @@ public function __invoke(DonationForm $form, Donation $donation, array $customFi
     {
         $form->schema()->walkFields(
             function (Field $field) use ($customFields, $donation) {
-                if (!array_key_exists($field->getName(), $customFields)) {
+                $fieldName = $field->getName();
+                $fieldType = $field->getType();
+
+                if (!array_key_exists($fieldName, $customFields)) {
                     return;
                 }
 
-                $value = $customFields[$field->getName()];
-
-                if ($field->shouldStoreAsDonorMeta()) {
-                    // save as donor meta
-                    give()->donor_meta->update_meta($donation->donorId, $field->getName(), $value);
+                /** @var File $field */
+                if (($fieldType === Types::FILE) && isset($_FILES[$fieldName])) {
+                    //TODO: let file field provide this storage logic
+                    $this->handleFileUpload($field, $donation);
                 } else {
-                    // save as donation meta
-                    give()->payment_meta->update_meta($donation->id, $field->getName(), $value);
+                    $value = $customFields[$fieldName];
+
+                    // TODO: it would be nice to have a way of serializing the value from the field api
+                    $value = apply_filters("givewp_store_custom_field_value_$fieldName", $value, $field, $donation);
+
+                    // TODO: it would be nice to have a way of customizing the storage from the field api
+                    do_action("givewp_store_custom_field_$fieldName", $value, $field, $donation);
```

Comment:
I plan on removing this in favor of a native fields api approach but wanted to showcase where something like this would be useful.  We recently discussed updates to the fields api to support serializing values and/or providing scopes for storage.

Reply:
I know what you're referring to regarding scopes, but I'm not sure what you're referring to for support for serializing values.

Reply:
I mean to say `sanitize` explained in [this](https://github.com/impress-org/givewp-next-gen/pull/236#discussion_r1278205373) comment